ogr/mock_core.py

```diff
@@ -0,0 +1,93 @@
+import functools
+import logging
+from unittest.mock import MagicMock
+from enum import IntEnum
+from ogr.abstract import PullRequest, PRComment, PRStatus
+import datetime
+
+logger = logging.getLogger(__name__)
+
+
+class MockType(IntEnum):
+    none = 1
+    read_only = 2
+    full = 3
+
+
+def readonly(_func=None, return_value=None, text="", use_internal_args=False):
+    def decorator_readonly(func):
+        @functools.wraps(func)
+        def readonly_func(self, *args, **kwargs):
+            read_only_suffix = "_read_only"
```

Comment:
What about using another parameter for the function -- `read_only_function=some.alternative.function` and use it in the `read_only` mode.

This is too much magic to me...;-)

```python
# some/alternative.py
def function(self, ...):
    return something
```
```python
@readonly(read_only_function=some.alternative.function)
def some_function(self, ...):
    return something
```

But I hope, we can stay with the `return_value` and only return some fake data.

But I am not deffinitelly against the `*_read_only`...;-) (I would only remove the state for these functions.)

Reply:
Why I've used this is, that I cannot pass self as parameter in decorator, What you are talking would be nice, but I have to pass also self parameter inside, what is more cryptic for me.
Super solution would be to use something like 
```
@readonly(read_only_function=self.alternative_function, pass_internal_values=True)
```
but I can use just what you said, something like what you suggested in inside decorador fuction do something like:
```
def readonly(_func,return_value=alternative_function, pass_internal_values=True, pass_object=True)
     ...
     if pass_object:
        return alternative_function(self, *args, **kwargs)
```